File: python_backend/getters/get_piece_type.py
from custom_except import *
from re import fullmatch


def get_piece_type(id_):
    """return 'Rook', 'Knight', 'Bishop', 'King', 'Queen', or 'Pawn'"""

    if id_ == '#':
        return 'Empty'
    if fullmatch(r'(W|B)R[1-9]', id_) is not None:
        return 'Rook'
    if fullmatch(r'(W|B)N[1-9]', id_) is not None:
        return 'Knight'
    if fullmatch(r'(W|B)B[1-9]', id_) is not None:
        return 'Bishop'
    if fullmatch(r'(W|B)Q[1-9]', id_) is not None:
        return 'Queen'
    if fullmatch(r'(W|B)K[1-9]', id_) is not None:
        return 'King'
    if fullmatch(r'(W|B)P[1-9]', id_) is not None:
        return 'Pawn'
    else:
        print('error: not a piece id')
        return -1


if __name__ == "__main__":

    for id in ['WP1', 'BB2', 'WQ3', 'WK1', 'BK1', 'WR2', 'WN1']:
        type_ = get_piece_type(id)
        print("id:{}, type:{}\n".format(id, type_))

File: react_frontend/src/sharedData/getPieceType.js

export function getPieceType(id_) {
    /*return ['Rook', 'Knight', 'Bishop', 'King', 'Queen', or] 'Pawn'**/
    if (id_ === '#') {
        return 'Empty'
    }
    if (id_.match(`^(W|B)R[1-9]$`) != null) {
        return 'Rook'
    }
    if (id_.match(`^(W|B)N[1-9]$`) != null) {
        return 'Knight'
    }
    if (id_.match(`^(W|B)B[1-9]$`) != null) {
        return 'Bishop'
    }
    if (id_.match(`^(W|B)Q[1-9]$`) != null) {
        return 'Queen'
    }
    if (id_.match(`^(W|B)K[1-9]$`) != null) {
        return 'King'
    }
    if (id_.match(`^(W|B)P[1-9]$`) != null) {
        return 'Pawn'
    }
    else {
        console.log('error: not a piece id')
        return -1
    }
}